src/state/migrations.ts

```diff
@@ -24,43 +14,20 @@ function tryParseOldState<T>(value: string | null, fallback: T): T {
 
 // The target version number is the key
 export const migrations: MigrationManifest = {
-  0: () => {
-    const oldTransactions = localStorage.getItem('redux_localstorage_simple_transactions')
-    const oldUser = localStorage.getItem('redux_localstorage_simple_user')
-    const oldLists = localStorage.getItem('redux_localstorage_simple_lists')
-    const oldSignatures = localStorage.getItem('redux_localstorage_simple_signatures')
-
-    const newTransactions = tryParseOldState(oldTransactions, initialTransactionsState)
-    const newUser = tryParseOldState(oldUser, initialUserState)
-    const newLists = tryParseOldState(oldLists, initialListsState)
-    const newSignatures = tryParseOldState(oldSignatures, {})
-
-    const result = {
-      user: legacyUserMigrations(newUser),
-      transactions: legacyTransactionMigrations(newTransactions),
-      lists: newLists,
-      signatures: newSignatures,
-      _persist: { version: 0, rehydrated: true },
-    }
-
-    localStorage.removeItem('redux_localstorage_simple_transactions')
-    localStorage.removeItem('redux_localstorage_simple_user')
-    localStorage.removeItem('redux_localstorage_simple_lists')
-    localStorage.removeItem('redux_localstorage_simple_signatures')
-    return result
-  },
+  0: migration0,
 }
 
 // We use a custom migration function for the initial state, because redux-persist
-// skips migration if there is no initial state in localStorage, but we want to migrate
+// skips migration if there is no initial state, but we want to migrate
 // previous persisted state from redux-localstorage-simple.
 export function customCreateMigrate(migrations: MigrationManifest, options: MigrationConfig): PersistMigrate {
   const defaultMigrate = createMigrate(migrations, options)
 
-  return (state: PersistedState, currentVersion: number) => {
+  return async (state: PersistedState, currentVersion: number) => {
     if (state === undefined) {
-      // If no state exists, run the migration for version 1
-      return Promise.resolve(migrations[0](undefined))
+      // If no state exists, run the legacy migration to set initial state
+      const initialState = await legacyLocalStorageMigration()
+      return defaultMigrate(initialState, currentVersion)
```

Comment:
This can be simplified with a fallthrough:

```suggestion
      // If no state exists, run the legacy migration to set initial state
      state = await legacyLocalStorageMigration()
```